src/numeric/processes.jl

```diff
@@ -8,36 +10,62 @@ abstract DiscreteProcess <: AbstractProcess
 abstract ContinuousProcess <: AbstractProcess
 abstract IIDExogenous <: AbstractProcess
 
-abstract AbstractDiscretizedProcess <: DiscreteProcess
+abstract AbstractDiscretizedProcess # <: DiscreteProcess
 
+###
+### Discretized process
+###
 
-# discretized process
-
+# date-t grid has a known structure
 type DiscretizedProcess <: AbstractDiscretizedProcess
-    nodes::Matrix{Float64}
-    integration_nodes::Array{Matrix{Float64}}
-    integration_weights::Array{Vector{Float64}}
+    grid::Grid
+    integration_nodes::Array{Matrix{Float64},1}
+    integration_weights::Array{Vector{Float64},1}
 end
 
-n_nodes(dp::DiscretizedProcess) = size(dp.nodes, 1)
+n_nodes(dp::DiscretizedProcess) = n_nodes(dp.grid)
+node(dp::DiscretizedProcess, i) = node(dp.grid,i)
 n_inodes(dp::DiscretizedProcess, i::Int) = size(dp.integration_nodes[i], 1)
-inodes(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_nodes[i][j, :]
-iweights(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_weights[i][j]
-node(dp::DiscretizedProcess, i) = dp.nodes[i, :]
+inode(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_nodes[i][j, :]
```

Comment:
I don't think we need IntegrationNodes objects. An integration node (a vector) is a feature of a discretized process, so we could directly do `dprocess[i,j]` or somthing alike. But this is syntactic sugar, so let's not do that.
The vector of matrices ? Look at the `x` argument of `residual` in https://github.com/EconForge/Dolo.jl/blob/master/src/algos/time_iteration.jl#L1 .
That one is of type `Array{Array{Float64,2},1}` : for each value of the exogenous shock, we have a matrix of values on the endogenous grid. Memory-wise this correspond to an Array{Float64,3} with dimensions `N_endo.n_x.N_exo` which is useful here because, we perform the extraction `x[i] ~ x[:,:,i]` many times, but inadequate for a human being: we want to think of a `N_exo.N_endo.n_x` array. 
So this is not perfect and not to be addressed in this PR. Essentially, I would like to stabilize user-facing functions first before revisiting all the memory issues. The solution might be a big dimension reversion, or some more levels of abstractions. But let's be super cautious here as iterating back and forth is not an option.